civis/io/_files.py

```diff
@@ -132,25 +135,18 @@ def _multipart_upload(buf, name, file_size, client, **kwargs):
     assert num_parts == len(urls), \
         "There are {} file parts but only {} urls".format(num_parts, len(urls))
 
-    # generate for writing a specific number of bytes from the buffer
-    def _gen_chunks(part_buf, max_bytes, chunk_size=32 * 1024):
-        bytes_read = 0
-        while True:
-            length = min(chunk_size, max_bytes - bytes_read)
-            if length <= 0:
-                break
-            bytes_read += length
-            data = part_buf.read(length)
-            yield data
-
     # upload function wrapped with a retry decorator
     @retry(RETRY_EXCEPTIONS)
-    def _upload_part_base(item, part_path):
+    def _upload_part_base(item, buf, part_size, file_size):
```

Comment:
You only need the file name here, correct? It's probably better to make that the argument instead of the whole buffer. That will avoid any possible thread safety or pickling issues when putting this function in a pool.